src/Compilers/Core/Portable/Compilation/SemanticModel.cs

```diff
@@ -876,5 +876,15 @@ protected internal virtual SyntaxNode GetTopmostNodeForDiagnosticAnalysis(ISymbo
         {
             return declaringSyntax;
         }
+
+        /// <summary>
+        /// Root of this semantic model
+        /// </summary>
+        internal SyntaxNode Root => RootCore;
+
+        /// <summary>
+        /// Root of this semantic model
+        /// </summary>
+        protected abstract SyntaxNode RootCore { get; }
```

Comment:
Perhaps `internal` rather than `protected` so it not included in the public API.